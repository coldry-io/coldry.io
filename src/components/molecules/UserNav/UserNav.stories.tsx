import type { Meta, StoryObj } from '@storybook/react';

import UserNav from '.';

export default {
  title: 'Molecules/UserNav',
  component: UserNav,
  parameters: {
    nextjs: {
      appDirectory: true
    }
  }
} as Meta<typeof UserNav>;

type Story = StoryObj<typeof UserNav>;

const user = {
  id: '1',
  image: 'https://randomuser.me/api/portraits/men/18.jpg',
  name: 'Brayden Olson',
  username: 'brayden.olson',
  email: 'brayden.olson@example.com'
};

export const Default: Story = {
  render: () => <UserNav user={user} />
};
