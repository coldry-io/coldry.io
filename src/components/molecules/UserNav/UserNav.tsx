'use client';

import { CreditCard, LayoutDashboard, LogOut, Settings, Users } from 'lucide-react';
import type { User } from 'next-auth';
import { signOut } from 'next-auth/react';
import { useRouter } from 'next/navigation';

import { Avatar, AvatarFallback, AvatarImage } from '@/components/atoms/Avatar';
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuGroup,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger
} from '@/components/atoms/DropdownMenu';

interface UserNavProps {
  user: User & {
    id: string;
    username?: string | null | undefined;
  };
}

const UserNav: React.FC<UserNavProps> = ({ user }) => {
  const router = useRouter();

  const getAvatarFallback = () => user.name?.substring(0, 1) ?? user.username?.substring(0, 1);

  return (
    <DropdownMenu>
      <DropdownMenuTrigger asChild>
        <Avatar className="h-8 w-8 cursor-pointer">
          {user.image ? (
            <AvatarImage src={user.image} alt={user.name ?? 'Profile Picture'} />
          ) : (
            <AvatarFallback className="bg-coldry-black text-coldry-white">
              {getAvatarFallback()}
            </AvatarFallback>
          )}
        </Avatar>
      </DropdownMenuTrigger>
      <DropdownMenuContent className="w-56" align="end" forceMount>
        <DropdownMenuLabel className="font-normal">
          <div className="flex flex-col space-y-1">
            <p className="text-sm font-medium leading-none">{user.name}</p>
            <p className="text-xs leading-none text-muted-foreground">{user.email}</p>
          </div>
        </DropdownMenuLabel>
        <DropdownMenuSeparator />
        <DropdownMenuGroup>
          <DropdownMenuItem onClick={() => router.push('/dashboard')} className="cursor-pointer">
            <LayoutDashboard className="mr-2 h-4 w-4" />
            <span>Dashboard</span>
          </DropdownMenuItem>
          <DropdownMenuItem
            onClick={() => router.push('/dashboard/team')}
            className="cursor-pointer"
          >
            <Users className="mr-2 h-4 w-4" />
            <span>Team</span>
          </DropdownMenuItem>
          <DropdownMenuItem
            onClick={() => router.push('/dashboard/billing')}
            className="cursor-pointer"
          >
            <CreditCard className="mr-2 h-4 w-4" />
            <span>Billing</span>
          </DropdownMenuItem>
          <DropdownMenuItem
            onClick={() => router.push('/dashboard/settings')}
            className="cursor-pointer"
          >
            <Settings className="mr-2 h-4 w-4" />
            <span>Settings</span>
          </DropdownMenuItem>
        </DropdownMenuGroup>
        <DropdownMenuSeparator />
        <DropdownMenuItem
          onClick={() => signOut({ callbackUrl: `${window.location.origin}/login` })}
          className="cursor-pointer"
        >
          <LogOut className="mr-2 h-4 w-4" />
          <span>Log out</span>
        </DropdownMenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  );
};

export default UserNav;
