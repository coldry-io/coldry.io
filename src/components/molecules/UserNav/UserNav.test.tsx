import { render } from '@/tests/test-utils';

import UserNav from '.';

jest.mock('next/navigation', () => require('next-router-mock'));

describe('UserNav', () => {
  it('should render the UserNav', () => {
    const user = {
      id: '1',
      image: 'https://randomuser.me/api/portraits/men/18.jpg',
      name: 'Brayden Olson',
      username: 'brayden.olson',
      email: 'brayden.olson@example.com'
    };

    const { asFragment } = render(<UserNav user={user} />);

    // TEST: should render the UserNav
    expect(asFragment()).toMatchSnapshot();
  });
});
