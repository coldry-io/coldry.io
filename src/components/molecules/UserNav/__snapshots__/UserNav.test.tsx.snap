// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNav should render the UserNav 1`] = `
<DocumentFragment>
  <span
    aria-expanded="false"
    aria-haspopup="menu"
    class="relative flex shrink-0 overflow-hidden rounded-full h-8 w-8 cursor-pointer"
    data-state="closed"
    id="radix-:r0:"
    type="button"
  />
</DocumentFragment>
`;
